fdbrpc/include/fdbrpc/FileTransfer.h

```diff
@@ -0,0 +1,134 @@
+/**
+ * FileTransfer.h
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2024 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifdef FLOW_GRPC_ENABLED
+#ifndef FDBRPC_FILE_TRANSFER_H
+#define FDBRPC_FILE_TRANSFER_H
+
+#include <optional>
+#include <fstream>
+#include <grpcpp/grpcpp.h>
+
+#include <boost/asio.hpp>
+
+#include "fdbrpc/file_transfer/file_transfer.grpc.pb.h"
+
+using fdbrpc::DownloadChunk;
+using fdbrpc::DownloadRequest;
+using fdbrpc::FileTransferService;
+using grpc::Channel;
+using grpc::ClientContext;
+using grpc::Status;
+
+class FileTransferServiceImpl final : public FileTransferService::Service {
+public:
+	grpc::Status DownloadFile(grpc::ServerContext* context,
+	                          const DownloadRequest* request,
+	                          grpc::ServerWriter<DownloadChunk>* writer) override {
+		std::ifstream input_file(request->file_name(), std::ios::binary | std::ios::ate);
+		if (!input_file.is_open()) {
+			return grpc::Status(grpc::StatusCode::NOT_FOUND, "File not found");
+		}
+
+		const std::streamsize file_size = input_file.tellg();
```

Comment:
If exception, file is closed?

Reply:
C++ library is RAII, so `ifstream` should close the file upon destruction which is what will happen when exception is raised.